include/swift/AST/AccessScope.h

```diff
@@ -48,7 +53,7 @@ class AccessScope {
   /// \see DeclContext::isChildContextOf
   bool isChildOf(AccessScope AS) const {
     if (!isPublic() && !AS.isPublic())
-      return getDeclContext()->isChildContextOf(AS.getDeclContext());
+      return allowsPrivateAccess(getDeclContext(), AS.getDeclContext());
```

Comment:
I'll update it to `allowsAccess` at the very least and will try to update `checkAccessibility`. I didn't like how there were multiple methods of doing this, but I was trying not to disturb too much. Let me see how it goes!